app/scenes/NewDocumentFromTemplate.tsx

```diff
@@ -0,0 +1,133 @@
+import { observer } from "mobx-react";
+import * as React from "react";
+import { useTranslation } from "react-i18next";
+import AutoSizer from "react-virtualized-auto-sizer";
+import { FixedSizeList as List } from "react-window";
+import styled from "styled-components";
+import { DocumentPath } from "~/stores/CollectionsStore";
+import Document from "~/models/Document";
+import Flex from "~/components/Flex";
+import { Outline } from "~/components/Input";
+import Labeled from "~/components/Labeled";
+import PathToDocument from "~/components/PathToDocument";
+import useSearchDocumentPath from "~/hooks/useSearchDocumentPath";
+import useStores from "~/hooks/useStores";
+import { newDocumentPath } from "~/utils/routeHelpers";
+
+type Props = {
+  template: Document;
+  onRequestClose: () => void;
+};
+
+function NewDocumentFromTemplate({ template, onRequestClose }: Props) {
+  const { collections } = useStores();
+  const { t } = useTranslation();
+  const { searchResult, setSearchTerm } = useSearchDocumentPath();
+
+  const handleFilter = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(ev.target.value);
+  };
+
+  const row = ({
+    index,
+    data,
+    style,
+  }: {
+    index: number;
+    data: DocumentPath[];
+    style: React.CSSProperties;
+  }) => {
+    const result = data[index];
+    return (
+      <PathToDocument
+        documentPath={result}
+        currentDocument={template}
+        collection={collections.get(result.collectionId)}
+        onClick={onRequestClose}
+        href={newDocumentPath(template.collectionId, {
```

Comment:
This jumps out of the react app and does a full page refresh, we should route programmatically or use `<Link to={....} />` to keep routing within the SPA